src/sgame/sg_bot_util.cpp

```diff
@@ -1935,7 +1935,7 @@ void BotFireWeaponAI( gentity_t *self )
 			// hopefully accounts for the movement of the bot and
 			// its target
 			constexpr float barbSafetyFactor = 5.0f/3.0f;
-			bool barbIsSafe = distance > (barbSafetyFactor * BG_Missile(MIS_BOUNCEBALL)->splashRadius);
+			bool barbIsSafe = distance > (barbSafetyFactor * BG_Missile(MIS_BOUNCEBALL)->splashRadius) || !self->botMind->botSkillSet[BOT_A_SAFE_BARBS];
```

Comment:
It's logically equivalent but it seems more intuitive to put the skill check as the first part of the OR. So that the other code is only executed if the skill is enabled. (This comment applies to both skills)

Reply:
I think this is more clear this way. You are thinking of writting it like this, right?

```
bool barbIsSafe = !self->botMind->botSkillSet[BOT_A_SAFE_BARBS] || distance > (barbSafetyFactor * BG_Missile(MIS_BOUNCEBALL)->splashRadius);
```

Reply:
Yes, like that.

Reply:
I find that confusing because you would think the condition is inverted. While it's actually that the bot *think* the barb is safe because it didn't bother to check that it actually isn't safe.

Reply:
Maybe I should change the variable name here.

Reply:
My line of thinking has nothing to do with barbs in particular; I think that optional calculations for skills should be gated behind the skill check.

Reply:
I mean most of the calculation is done anyway 🤷‍♂️️